keeper/src/main/scala/zio/keeper/errors.scala

```diff
@@ -0,0 +1,56 @@
+package zio.keeper
+
+import zio.duration.Duration
+import zio.nio.SocketAddress
+
+import scala.reflect.ClassTag
+
+sealed abstract class Error(val msg: String = "")
+
+sealed abstract class SerializationError(msg: String = "") extends Error(msg = msg)
+
+object SerializationError {
+
+  final case class SerializationTypeError[A](val cause: Throwable)(implicit ct: ClassTag[A])
```

Comment:
I'm not super happy with us using class tags as they will bleed through everywhere we use this in a generic context

Reply:
I'm not happy either. Let me know how we can improve that.